staging/src/k8s.io/pod-security-admission/policy/checks.go

```diff
@@ -80,6 +82,19 @@ type CheckResult struct {
 	// - list specific invalid containers: "container1, container2"
 	// - list specific non-default capabilities: "CAP_NET_RAW"
 	ForbiddenDetail string
+	// ErrList must be set if Allowed is false.
+	// ErrList is a detailed list of restrained field errors.
+	ErrList field.ErrorList
+}
+
+// RestrictedFieldErr holds the detail information about restricted field error.
+type RestrictedFieldErr struct {
```

Comment:
I don't think you need this type. The field validation package should have all the types you need for the errors.

Reply:
I think we need this type because we not only need a restricted field path, but we also need the `containerName` and `ForbiddenValues` to exclude images and values.

For example, in Kyverno, user can create a ClusterPolicy like this if they want to apply Restricted PSS but to skip checking the Control "Privilege Escalation" for pods running "nginx*" or "redis*". And that's exactly why we need the `containerName` field.
```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: psa
spec:
  background: true
  validationFailureAction: Enforce
  rules:
  - name: restricted
    match:
      any:
      - resources:
          kinds:
          - Pod
    validate:
      podSecurity:
        level: restricted
        version: latest
        exclude:
        - controlName: Privilege Escalation
          images:
          - nginx*
          - redis*
```
Another example is that user can create a Cluster like this if they want to apply Restricted PSS but to skip checking the Control "Capabilities" for pods running "nginx*" or "redis*" to be able to add `KILL`, `SETGID` and `SETUID` capabilities. And that's exactly why we need the `ForbiddenValues` field.
```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: psa
spec:
  background: true
  validationFailureAction: Enforce
  rules:
  - name: restricted
    match:
      any:
      - resources:
          kinds:
          - Pod
    validate:
      podSecurity:
        level: restricted
        version: latest
        exclude:
        - controlName: Capabilities
          images:
          - nginx*
          - redis*
          values:
          - "KILL"
          - "SETGID"
          - "SETUID"
```
In summary, we need this type to store the information we want and place it in the badValue field of the field.Error type.